tests/mfx_dispatch/linux/mfx_dispatch_test_main.cpp

```diff
@@ -19,33 +19,36 @@
 // SOFTWARE.
 
 #include "gtest/gtest.h"
-#include <mfxvideo.h>
+#include "mfx_dispatch_tests_main.h"
+
+DlopenHookType g_e_dlopen_hook_state = DlopenHookType::ALWAYS_NULL;
+DlsymHookType  g_e_dlsym_hook_state  = DlsymHookType::ALWAYS_NULL;
+
+std::map<DlopenHookType, DlopenHook> g_dlopen_hook_table {
+            {DlopenHookType::ALWAYS_NULL, TEST_DLOPEN_HOOKS::AlwaysNull},
+    };
+std::map<DlsymHookType,  DlsymHook>  g_dlsym_hook_table {
+            {DlsymHookType::ALWAYS_NULL, TEST_DLSYM_HOOKS::AlwaysNull},
+    };
+
 
 extern "C"
 {
   void *dlopen(const char *filename, int flag)
   {
     printf("dlopen: filename=%s, flag=%d\n", filename, flag);
-    return NULL;
+    return g_dlopen_hook_table[g_e_dlopen_hook_state](filename, flag);
```

Comment:
I think you need to implement a range of tests to cover minimal variety of possible tests. This may cause dramatic change to your current idea of 2 maps for dlopen and dlsym. For example, one of the dispatcher tasks is to load the library of the lower version even if dispatcher was built with the higher version. Thus, you will have situation when you will need to return valid pointers in dlsym for 0..M uAPI functions and return NULL for M+1..N. Think how you will handle this.